### ISSUE TITLE
Todo looks different from note in LaTeX output

### ISSUE THREAD

**Issue by jessetan** [sphinx-doc/sphinx#3725]
### Problem
For HTML output, the `todo` directive looks the same as a `note` with a different title. This is OK:
![todo-html](https://cloud.githubusercontent.com/assets/3284111/25795274/ed6b5a38-33d4-11e7-97f9-1764fe23462d.png)

For LaTeX output, the contents of the `todo` directive is placed on a new line instead of next to the title and is missing the colon:
![todo-latex](https://cloud.githubusercontent.com/assets/3284111/25795297/07516000-33d5-11e7-8641-7c69503052ae.png)

The generated LaTeX code is:
``` latex
\begin{sphinxadmonition}{warning}{Warning:}
This is a warning
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Note:}
This is a note
\end{sphinxadmonition}

\begin{sphinxadmonition}{note}{Todo}

This is a todo
\end{sphinxadmonition}
```

#### Procedure to reproduce the problem

Use the following rst input and build HTML and PDF:

``` rst
.. warning:: This is a warning

.. note:: This is a note

.. todo:: This is a todo
```

#### Expected results
I expect the `todo` directive to have the same output as a `note`. The generated LaTeX should be:
``` latex
\begin{sphinxadmonition}{note}{Todo:}
This is a todo
\end{sphinxadmonition}
```

This means removing the empty line which causes the content to become a new paragraph and adding the colon in the title.

### Environment info
- OS: OSX 10.10.5
- Python version: 2.7.13
- Sphinx version: 1.5.5
- docutils 0.13.1
- TeX Live 2016

**Comment by jfbu** [comment]
The issue here is that `todo`  type of admonition is not known to LaTeX writer. And there is a difficulty that `todo` extension always adds a title to the todo node. Here is a horrible hack which I am indicating only with the hope Python maintainers fix it:

```txt
diff --git a/sphinx/ext/todo.py b/sphinx/ext/todo.py
index cd35631..46aca12 100644
--- a/sphinx/ext/todo.py
+++ b/sphinx/ext/todo.py
@@ -191,6 +191,15 @@ def depart_todo_node(self, node):
     self.depart_admonition(node)
 
 
+def visit_latex_todo_node(self, node):
+    print(node)
+    self.body.append(u'\n\\begin{sphinxadmonition}{note}{Todo:}')
+
+
+def depart_latex_todo_node(self, node):
+    self.body.append('\\end{sphinxadmonition}\n')
+
+
 def setup(app):
     app.add_event('todo-defined')
     app.add_config_value('todo_include_todos', False, 'html')
@@ -200,7 +209,7 @@ def setup(app):
     app.add_node(todolist)
     app.add_node(todo_node,
                  html=(visit_todo_node, depart_todo_node),
-                 latex=(visit_todo_node, depart_todo_node),
+                 latex=(visit_latex_todo_node, depart_latex_todo_node),
                  text=(visit_todo_node, depart_todo_node),
                  man=(visit_todo_node, depart_todo_node),
                  texinfo=(visit_todo_node, depart_todo_node))
diff --git a/sphinx/writers/latex.py b/sphinx/writers/latex.py
index be0771c..80468bb 100644
--- a/sphinx/writers/latex.py
+++ b/sphinx/writers/latex.py
@@ -823,7 +823,10 @@ class LaTeXTranslator(nodes.NodeVisitor):
 
     def visit_title(self, node):
         parent = node.parent
-        if isinstance(parent, addnodes.seealso):
+        # horrible mais ça marche
+        if str(type(parent))=="<class 'sphinx.ext.todo.todo_node'>":
+            raise nodes.SkipNode
+        elif isinstance(parent, addnodes.seealso):
             # the environment already handles this
             raise nodes.SkipNode
         elif isinstance(parent, nodes.section):
```

the patch is on 1.5.5.

One could also find a workaround by some LaTeX macros you could add to your conf.py (in `latex_elements['preamble']`) to modify sphinxadmonition environment to detect that it is used with `{note}{Todo}` and act appropriately.

This being all said, Todo notes are not for final document, so is it really a problem if they are not rendered like other admonitions ?

**Comment by jfbu** [comment]
in patch from previous comment there is a `print(node)` which I used in debugging: it should be removed of course.

**Comment by jfbu** [comment]
For a workaround at latex run time, add this to `conf.py`

```
latex_elements = {
    'preamble': r"""
\makeatletter
\def\sphinx@todo@title{Todo}
\renewenvironment{sphinxnote}[1]
  {\begin{sphinxlightbox}%
   \def\@tempa{#1}%
   \ifx\@tempa\sphinx@todo@title
       \expandafter\@firstoftwo\else\expandafter\@secondoftwo
   \fi
   {\sphinxstrong{Todo:} \space\@gobble}{\sphinxstrong{#1} }}
  {\end{sphinxlightbox}}
\makeatother
""",
}
```

Caveat: you must replace "Todo" in the macro `\sphinx@todo@title` by the string suitable for the language of the document. But yours in English obviously, so simply "Todo".

This will keep the tex file as in your report but the PDF will look as desired.

By the way, it is a legacy fact that there are two space tokens after the prefix `Note:` or other "light" admonitions. When admonitions were refactored at e5db9da this was kept. One space token comes from the definitions of the `sphinxnote`, `sphinxhint` etc... environments, and the other one comes from the end of the line containing `\begin{sphinxnote}`, etc...

**Comment by jfbu** [comment]
if any French person passes by, there is a complication that the `:` as in `Todo:` in the latex macros above must be made "active", else Babel will not insert the automatic spacing before it. If you don't know what I am talking about you are probably not French, and if you do know, then yu also know what I mean and what to do additionally to the macros in my comment.

**Comment by jessetan** [comment]
> This being all said, Todo notes are not for final document, so is it really a problem if they are not rendered like other admonitions ?

No it is just a minor inconsistency issue. I found it when I wanted to remove the colon from the title in LaTeX output by just removing the last character of the title, but then "Todo" becomes "Tod" and our writers search documents for "todo" to find things to work on.

**Comment by jfbu** [comment]
Does your search need to be done on the tex file or the pdf? if the latter does the LaTeX workaround I indicated work for your document?

**Comment by tk0miya** [comment]
Fixed at #3733.
Thank you for reporting!
